### DISCUSSION TITLE
Octopus Pro with DWIN Display

### DISCUSSION THREAD

**Discussion by gsuresh2u** [mriscoc/Ender3V2S1#895] [Q&A] [answered]
Hi, i want to upgrade my Ender3 V2 with Skr Mini E3V3 to Octopus Pro
but i want tto use DWIN display which came wih Ender3 V2
i want to use G34 ( auto Z alien)
is there any way to use this ?
Plese check https://github.com/MarlinFirmware/Marlin/pull/24983
@mriscoc , Please look into this, i want to use this firmware, buying new display is no problem
but i want to DWIN display, i will contribute or Sponser for this Octopus Board for your Firmware with DWIN display
Please look into tthis

**Comment by mriscoc** [comment]
Hi, were you able to use Marlin bugfix with that board and DWIN (non-touch)?

**Comment by gsuresh2u** [comment]
Not tried yet, I have octopus pro board
I want to use your firmware, from the link above, it's possible to use DWIN display with this board
Am I right?

**Comment by mriscoc** [comment]
I found this file to bring support for the BTT Octopus V1.1, extract and save it in the _boards folder from Special Configuration repository
[OCTOPUSV1_1.zip](https://github.com/mriscoc/Ender3V2S1/files/11521564/OCTOPUSV1_1.zip)

**Comment by gsuresh2u** [comment]
@mriscoc , today i tried to connect Ender3 v2 display to Octopus Pro 1.0
i got success, getting display fine,please check the below image
![Fine](https://github.com/mriscoc/Ender3V2S1/assets/39871361/288b1ed0-be01-4c51-97d9-8ebe622074cc)

 but when i try to enable ProUI features, i am getting error as below image
![Error](https://github.com/mriscoc/Ender3V2S1/assets/39871361/dc13e1b4-d4aa-439b-8a7f-965fbe7b2434)

is there anyway to enable ProUI ?
other problem i found is, if sd card is in slot before power on to board, its showing sd card gcodes well
but if sd card removed and inserted other card, its still showing old sd card contents, until i need to restart board
and USB drive also not working, i am attaching my configs, Please check, i will contribute

[Configs.zip](https://github.com/mriscoc/Ender3V2S1/files/12711621/Configs.zip)

**Reply by mriscoc** [reply]
Hi, the error about the architecture is because the script can't find a pre-build library that fits your board's processor, what environment are you using? 
For the error about SD card, that is because there is an error in the SD card detection pin, or your card slot doesn't have a card detection switch, in that case enable `HAS_SD_EXTENDER`, that will disable the card detection and forces a media update each time you open the media menu.

**Reply by gsuresh2u** [reply]
Sir, I am using STM32F429ZG_btt_USB
About using ProUI, there is possibility for this board?

**Reply by mriscoc** [reply]
Try adding this to the build_flags: `MCU_STM32F429` of that environment.

**Reply by gsuresh2u** [reply]
Added that, same environment error
added like below
```
[env:STM32F429ZG_btt_USB]
extends           = env:STM32F429ZG_btt
platform_packages = ${stm_flash_drive.platform_packages}
build_unflags     = -DUSBD_USE_CDC
build_flags       = ${stm_flash_drive.build_flags}
                    -DUSE_USB_HS_IN_FS -DUSE_USBHOST_HS
                    -DUSBD_IRQ_PRIO=5 -DUSBD_IRQ_SUBPRIO=6
                    -DUSBD_USE_CDC_MSC -MCU_STM32F429
```

**Reply by mriscoc** [reply, marked as answer]
The correct sintaxis is -DMCU_STM32F429 (-D means define)

**Reply by gsuresh2u** [reply]
Sir, its working great, got tool bar also
Thanks for sugession
usb port is working fine, but sd card slot is not working
sd card slot only working for update firmware
i am attaching my configs, please check
[Configs.zip](https://github.com/mriscoc/Ender3V2S1/files/12723421/Configs.zip)

**Reply by mriscoc** [reply]
Get it working first with the current Marlin bugfix branch.

**Comment by ubeyou** [comment]
Hi there, are you able to provide the pin connection from the board to the DWIN? I'm using SKR 3 and not sure whats the pinout for it.

**Reply by gsuresh2u** [reply]
Hi, plz check https://github.com/MarlinFirmware/Marlin/pull/24983
